File: 프로그래머스/2/42578. 의상/의상.py
from collections import *

def solution(clothes):
    answer = 0
    
    clothes_dict = defaultdict(int)
    
    for _, category in clothes:
        clothes_dict[category] += 1
    
    answer = 1

    for count in clothes_dict.values():
        answer *= (count + 1)

    
    
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        /*
        
        */
        HashMap<String, Integer> map = new HashMap<>(); 
        
        for (String[] cloth : clothes) {
            String type = cloth[1];
            
            map.put(type, map.getOrDefault(type, 0) + 1);
        }
        
        int answer = 1;
        
        for (Integer value : map.values()) {
            answer *= (value + 1);
        }
        
        return answer - 1;
    }
}